megamock/import_machinery.py

```diff
@@ -68,21 +114,24 @@ def new_import(*args, **kwargs) -> ModuleType:
             and len(args) > 3
             and (names := args[3])
         ):
-            stack = inspect.stack()
-            for frame in stack:
-                if frame.code_context is None:
-                    continue
-                if frame.function == "new_import":
+            for i in range(1, 5):
+                frame = sys._getframe(i)
+                if frame.f_code.co_name == "new_import":
                     continue
-                calling_module = inspect.getmodule(frame[0])
+                calling_module = inspect.getmodule(frame)
                 if calling_module:
                     break
             assert calling_module
             full_line = _reconstruct_full_line(frame)
             for k in names:
                 if full_line and (
-                    renamed_result := re.search(
-                        rf"\s*from \S+ import.*{k}\s+as\s+(\w+)", full_line, re.DOTALL
+                    has_rename_regex.search(full_line)
+                    and (
+                        renamed_result := re.search(
+                            rf"\s*from \S+ import.*{k}\s+as\s+(\w+)",
+                            full_line,
+                            re.DOTALL,
+                        )
```

Comment:
This appears to be a test gap. If `{k}` is removed, all tests pass, but it should be necessary. Maybe something like:

`from ... import var_a, var_b as var_bee` would catch this?